src/cairoUtilFuncGen/abi/abiEncode.ts

```diff
@@ -342,15 +342,7 @@ export class AbiEncode extends AbiBase {
   }
 
   private createArrayInlineEncoding(type: ArrayType) {
-    let key = 'inline ' + type.pp();
-    // Modifiy key to generate same function for diferent size (but same element type)
-    // static arrays
-    if (type.size !== undefined) {
-      const reversedKey = key.split('').reverse().join('');
-      // swapping '[<num>]' for '[]' but since string is reversed we are swapping ']<num>[' for ']['
-      const parsedKey = reversedKey.replace(/\][0-9]+\[/, '][');
-      key = parsedKey.split('').reverse().join('');
-    }
+    const key = 'inline ' + (type.size !== undefined ? type.pp() : removeSizeInfo(type));
```

Comment:
In order to execute the function call `removeSizeInfo(type)`, the condition `type.size !== undefined` must be false. Then the first statement of this function is an assert with `type.size !== undefined` as the condition, which means that everytime this function call is executed, it throws an assertion error. Is this what was intended? I'm asking because that's not what was happening in the code that was removed.

Reply:
Good find, I overlooked. This wouldn't make the code to break but to generate extra functions